WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductImageThumbnail.swift

```diff
@@ -7,27 +7,39 @@ struct ProductImageThumbnail: View {
     private let scale: CGFloat
     private let productImageCornerRadius: CGFloat
     private let foregroundColor: Color
+    private let cache: ImageCache = ImageCache.default
+    private let usesDefaultCache: Bool
 
     /// Image processor to resize images in a background thread to avoid blocking the UI
     ///
     private var imageProcessor: ImageProcessor {
-        ResizingImageProcessor(referenceSize:
-                .init(width: productImageSize * scale,
-                      height: productImageSize * scale),
-                               mode: .aspectFill
-        )
+        if usesDefaultCache {
+            DefaultImageProcessor()
+        } else {
+            ResizingImageProcessor(referenceSize:
+                    .init(width: productImageSize * scale,
+                          height: productImageSize * scale),
+                                   mode: .aspectFill)
+        }
```

Comment:
The current usage of `ResizingImageProcessor` breaks KF's built-in cache, if we use this processor it only seems to cache very specific images, but most of them are not cached. This is resolved as soon as we pass the `DefaultImageProcessor`. 

I've also confirmed that by removing the processor, other places where the `ProductImageThumbnail` is used (like during product selection on Order creation) images also start to be cached automatically, which currently doesn't happen, making for a better experience. Since this is outside of the scope of the PR and would add many testing cases related to the app and not the POS I've left it outside for the moment (this was added initially to tackle stores with multiple and large images), so we keep using the existing implementation in the app, but the default processor in POS.